README.md

```diff
@@ -15,17 +15,19 @@ The example located in `test/Main.hs` showcases a consumer & producer running co
 
 ```haskell
 main :: IO ()
-main = runPulsar resources $ \(Consumer {..}, Producer {..}) ->
+main = runPulsar conn $ do
+  c <- newConsumer topic "test-sub"
+  p <- newProducer topic
+  liftIO $ program c p
+
+conn :: PulsarConnection
+conn = connect defaultConnectData
+
+program :: Consumer IO -> Producer IO -> IO ()
+program (Consumer fetch ack) (Producer send) =
   let c = forever $ fetch >>= \(Message i m) -> msgDecoder m >> ack i
-      p = forever $ sleep 5 >> traverse_ produce messages
+      p = forever $ sleep 5 >> traverse_ send messages
```

Comment:
why this sleep is needed?

Reply:
@psisoyev you need to sleep FFS :smile:

Reply:
That `sleep` is just to generate messages every 5 seconds, forever. It makes for a better example IMO but it is not needed.

Reply:
shouldn't we send out the message as soon as we have it?

Reply:
We could but we need to guarantee we are consuming before start sending events and that cannot be guaranteed by using `concurrently` this way, it is a very naive program...

Reply:
There are still a few rough edges I need to work through to make it more robust, this PR is a step towards that goal